app/login/page.tsx

```diff
@@ -1,15 +1,6 @@
-import { auth } from '@/auth'
 import LoginForm from '@/components/login-form'
-import { Session } from '@/lib/types'
-import { redirect } from 'next/navigation'
 
 export default async function LoginPage() {
-  const session = (await auth()) as Session
-
-  if (session) {
-    redirect('/')
-  }
```

Comment:
it's better to centralize this logic of public vs protected pages in the middleware.